// Copyright 2022 ReductStore
// This Source Code Form is subject to the terms of the Mozilla Public
//    License, v. 2.0. If a copy of the MPL was not distributed with this
//    file, You can obtain one at https://mozilla.org/MPL/2.0/.

syntax = "proto3";
import "reduct/proto/api/bucket.proto";

package reduct.proto.api;

message Defaults {
  BucketSettings bucket = 1;  // default values for a new bucket
}

// Stats of server
message ServerInfo {
  string version = 1;       // version of storage
  uint64 bucket_count = 2;  // number of buckets in storage
  uint64 usage = 3;         // disk usage in bytes
  uint64 uptime = 4;        // uptime in seconds
  uint64 oldest_record = 5; // unix timestamp of oldest record in microseconds
  uint64 latest_record = 6; // unix timestamp of latest record in microseconds

  Defaults defaults = 7;    // default settings for everything
}